app/src/main/java/edu/unikom/herbamedjabar/repository/PlantRepository.kt

```
@@ -4,67 +4,116 @@ import android.app.Application
-                } ?: throw Exception("Hasil teks dari AI kosong.")
-
-            } catch (e: Exception) {
-                currentRetry++
-                if (currentRetry >= maxRetries) {
+                } catch (e: androidx.sqlite.SQLiteException) {
+                    savedImagePath?.let { runCatching { File(it).delete() } }
+                    throw e
+                } catch (e: IllegalArgumentException) {
+                    savedImagePath?.let { runCatching { File(it).delete() } }
+                    throw e
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)